gateway-js/src/index.ts

```diff
@@ -551,28 +640,27 @@ export class ApolloGateway implements GraphQLService {
       this.logger.info('Updated Supergraph SDL was found.');
     }
 
-    await this.maybePerformServiceHealthCheck(result);
+    await this.maybePerformServiceHealthCheck({ supergraphSdl, id });
```

Comment:
If this throws then the `update` function throws. Is that what we want? (Maybe it is, just need to be sure that the example code for this API takes that into account.)

Alternatively: would it make sense to separate "update with SDL" from "run health checks against services in SDL"? eg, these could be two different hooks passed to the supergraphSdl function (`update` and `healthCheck`) and it's your job to call both if you want, or perhaps the latter could just be a public method on the gateway.  (This would perhaps let us make `update` synchronous which would let us stop worrying about it being called concurrently though we still have to worry about it being called re-entrantly.)

Reply:
Did all of the above, this was a nice refactor. `maybeHealthCheck` is now pulled up and awaited before either update function is called, allowing those to both be sync. I'm now providing a `healthCheck` hook to the function alongside update.